app/uk/gov/hmrc/gform/gform/DeclarationController.scala

```diff
@@ -65,40 +66,46 @@ class DeclarationController(
   def showDeclaration(
     maybeAccessCode: Option[AccessCode],
     formTemplateId: FormTemplateId,
-    suppressErrors: SuppressErrors
+    suppressErrors: SuppressErrors,
+    maybeCoordinates: Option[Coordinates],
+    taskCompleted: Option[Boolean]
```

Comment:
I don't think we should be reusing functionality for `declarationSection` to process task list declaration pages, they are not real declarations. 
We should have bespoke implementation for task list declaration pages logic.

I'm fine if you want to reuse `case class DeclarationSection`, but apart from that let's not mix these two.